### ISSUE TITLE
Lora finetuned model shows NaN

### ISSUE THREAD

**Issue by jellyho** [openvla/openvla#53]
I've finetuned model using lora, and the results graph is shown like this.
![image](https://github.com/user-attachments/assets/96faaf7f-2e07-4a3c-8e4d-3289f708bfe9)

So I tried to inference the model, and I found that my model generates same value regardless of input.
It was weird, so I tried vla forward function and I could see that the loss came out nan.

Why does nan appear in reference even though the learning results are good? I think there may be a problem in loading the pretrained checkpoint, but is there a hint that comes to mind?

Below codes are the way I loaded pretrained model. I didn't used quantization for train.
```python3
ckpt_dir = '/home/jellyho/openvla_run/openvla-7b+bm_sim+b16+lr-0.0005+lora-r32+dropout-0.0'
dataset_stats_file =  open(ckpt_dir + '/dataset_statistics.json', 'r')
norm_stats = json.load(dataset_stats_file)
# print('norm_stats', norm_stats)
AutoConfig.register("openvla", OpenVLAConfig)
AutoImageProcessor.register(OpenVLAConfig, PrismaticImageProcessor)
AutoProcessor.register(OpenVLAConfig, PrismaticProcessor)
AutoModelForVision2Seq.register(OpenVLAConfig, OpenVLAForActionPrediction)

# Load Processor & VLA
processor = AutoProcessor.from_pretrained(ckpt_dir, trust_remote_code=True)
vla = AutoModelForVision2Seq.from_pretrained(
    ckpt_dir, # [Optional] Requires `flash_attn`
    torch_dtype=torch.bfloat16,
    low_cpu_mem_usage=True,
    quantization_config=None,
    trust_remote_code=True,
    norm_stats=norm_stats
).to("cuda:0")
```

and this is the command I used for run finetuning
```bash
torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py \
  --vla_path "openvla/openvla-7b" \
  --data_root_dir /home/jellyho/tensorflow_datasets \
  --dataset_name bm_sim \
  --run_root_dir /home/jellyho/openvla_run \
  --adapter_tmp_dir /home/jellyho/openvla_tmp \
  --lora_rank 32 \
  --batch_size 4 \
  --grad_accumulation_steps 4 \
  --learning_rate 5e-4 \
  --image_aug True \
  --wandb_project OpenVLA \
  --wandb_entity jellyho_ \
  --save_steps 1000
```

**Comment by jellyho** [comment]
I found that after running merge_and_unload(), some of the parameter became NaN.

```
base_vla = AutoModelForVision2Seq.from_pretrained(
                        'openvla/openvla-7b', torch_dtype=torch.bfloat16, low_cpu_mem_usage=True, trust_remote_code=True
)
merged_vla = PeftModel.from_pretrained(base_vla, '/home/jellyho/openvla_tmp/openvla-7b+bm_sim+b16+lr-0.0005+lora-r32+dropout-0.0')

for name, param in merged_vla.named_parameters():
    if torch.isnan(param).any():
        print(f"Parameter {name} contains NaN values.")
        print(f"Parameter {name}: min={param.min().item()}, max={param.max().item()}")
```

If I do this, there are no NaN value in parameters. But after I run .merge_and_unload(),

```
merged_vla = merged_vla.merge_and_unload()
for name, param in merged_vla.named_parameters():
    if torch.isnan(param).any():
        print(f"Parameter {name} contains NaN values.")
        print(f"Parameter {name}: min={param.min().item()}, max={param.max().item()}")
```
```
Parameter vision_backbone.featurizer.blocks.19.attn.proj.weight contains NaN values.
Parameter vision_backbone.featurizer.blocks.19.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.featurizer.blocks.20.attn.proj.weight contains NaN values.
Parameter vision_backbone.featurizer.blocks.20.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.featurizer.blocks.21.attn.proj.weight contains NaN values.
Parameter vision_backbone.featurizer.blocks.21.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.featurizer.blocks.22.attn.proj.weight contains NaN values.
Parameter vision_backbone.featurizer.blocks.22.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.featurizer.blocks.23.attn.proj.weight contains NaN values.
Parameter vision_backbone.featurizer.blocks.23.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.fused_featurizer.blocks.9.attn.proj.weight contains NaN values.
Parameter vision_backbone.fused_featurizer.blocks.9.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.fused_featurizer.blocks.10.attn.proj.weight contains NaN values.
Parameter vision_backbone.fused_featurizer.blocks.10.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.fused_featurizer.blocks.11.attn.proj.weight contains NaN values.
Parameter vision_backbone.fused_featurizer.blocks.11.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.fused_featurizer.blocks.12.attn.proj.weight contains NaN values.
Parameter vision_backbone.fused_featurizer.blocks.12.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.fused_featurizer.blocks.13.attn.proj.weight contains NaN values.
Parameter vision_backbone.fused_featurizer.blocks.13.attn.proj.weight: min=nan, max=nan
Parameter vision_backbone.fused_featurizer.blocks.14.attn.proj.weight contains NaN values.
Parameter vision_backbone.fused_featurizer.blocks.14.attn.proj.weight: min=nan, max=nan
```

The parameter of vision_backbone became NaN. This is why I got NaN output from the model that loaded the (merge_and_unloaded) checkpoint.
